crates/bevy_reflect/src/map.rs

```diff
@@ -243,3 +251,25 @@ pub fn map_partial_eq<M: Map>(a: &M, b: &dyn Reflect) -> Option<bool> {
 
     Some(true)
 }
+
+/// Applies the elements of `b` to the corresponding elements of `a`.
+///
+/// If a key from `b` does not exist in `a`, the value is cloned and inserted.
+///
+/// # Panics
+///
+/// This function panics if `b` is not a map.
```

Comment:
Nit: Maybe reiterate that these two parameters are maps (as opposed to an iterator of tuples or something else).
```suggestion
/// Applies the elements of reflected map `b` to the corresponding elements of map `a`.
///
/// If a key from `b` does not exist in `a`, the value is cloned and inserted.
///
/// # Panics
///
/// This function panics if `b` is not a reflected map.
```

> Feel free to change my wording if preferred.